packages/world/src/modules/keyswithvalue/KeysWithValueModule.sol

```diff
@@ -23,43 +24,55 @@ import { getTargetTableSelector } from "../utils/getTargetTableSelector.sol";
  *
  * Note: if a table with composite keys is used, only the first key is indexed
  *
- * Note: this module currently expects to be `delegatecalled` via World.installRootModule.
- * Support for installing it via `World.installModule` depends on `World.callFrom` being implemented.
+ * Note: this module currently only supports `installRoot` (via `World.installRootModule`).
+ * TODO: add support for `install` (via `World.installModule`) by using `callFrom` with the `msgSender()`
  */
 contract KeysWithValueModule is IModule, WorldContextConsumer {
   using ResourceSelector for bytes32;
 
   // The KeysWithValueHook is deployed once and infers the target table id
   // from the source table id (passed as argument to the hook methods)
-  KeysWithValueHook immutable hook = new KeysWithValueHook();
+  KeysWithValueHook private immutable hook = new KeysWithValueHook();
 
   function getName() public pure returns (bytes16) {
     return bytes16("index");
   }
 
-  function install(bytes memory args) public override {
+  function installRoot(bytes memory args) public {
     // Extract source table id from args
     bytes32 sourceTableId = abi.decode(args, (bytes32));
     bytes32 targetTableSelector = getTargetTableSelector(MODULE_NAMESPACE, sourceTableId);
 
+    IBaseWorld world = IBaseWorld(_world());
+
     // Register the target table
-    KeysWithValue.register(IBaseWorld(_world()), targetTableSelector);
+    KeysWithValue.register(world, targetTableSelector);
 
     // Grant the hook access to the target table
-    IBaseWorld(_world()).grantAccess(targetTableSelector, address(hook));
+    world.grantAccess(targetTableSelector, address(hook));
 
     // Register a hook that is called when a value is set in the source table
-    StoreSwitch.registerStoreHook(
-      sourceTableId,
-      hook,
-      StoreHookLib.encodeBitmap({
-        onBeforeSetRecord: true,
-        onAfterSetRecord: false,
-        onBeforeSetField: true,
-        onAfterSetField: true,
-        onBeforeDeleteRecord: true,
-        onAfterDeleteRecord: false
-      })
+    (bool success, bytes memory returnData) = address(world).delegatecall(
+      abi.encodeCall(
+        world.registerStoreHook,
+        (
+          sourceTableId,
+          hook,
+          StoreHookLib.encodeBitmap({
+            onBeforeSetRecord: true,
+            onAfterSetRecord: false,
+            onBeforeSetField: true,
+            onAfterSetField: true,
+            onBeforeDeleteRecord: true,
+            onAfterDeleteRecord: false
+          })
+        )
+      )
     );
+    if (!success) revertWithBytes(returnData);
```

Comment:
would it make sense to move this pattern into a lib for the commonly used methods for ergonomics?

```solidity
WorldRoot.registerStoreHook(...)
```

Reply:
agree it would be more ergonomic to use, but unfortunately more gas expensive because of all the parameters that need to be passed around in memory, and that library would be pretty painful to maintain (because it would have to include all the function signatures of all systems of `IBaseWorld` which includes all the functions the `CoreModule` registers)